include/gridtools/stencil_composition/sid/block.hpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include "../../common/generic_metafunctions/for_each.hpp"
+#include "../../common/host_device.hpp"
+#include "../../common/hymap.hpp"
+#include "../../common/tuple_util.hpp"
+#include "../../meta.hpp"
+#include "./concept.hpp"
+#include "./delegate.hpp"
+
+namespace gridtools {
+    namespace sid {
+        template <class Dim>
+        struct blocked_dim {
+            using type = blocked_dim;
+        };
+
+        namespace block_impl_ {
+            template <class Stride, class BlockSize>
+            struct blocked_stride {
```

Comment:
I would use `tuple` here to benefit from empty base optimization.
It matters because strides pass through the host/device boundary